Elynor-Kamil/2021-Emergency-Management-System
# Interface integration

Since this PR will integrate with all the controller functions, I think it's better if everyone can at least have a look on and test the parts that link to the controller functions they're responsible for. Please let me know if you've spotted errors on the front-end and I'd be happy to hear any suggestion on the interface design!

### Main Updates:
- `cli.py` includes EmsShell(), the parent class of all other shells. It validates user credentials and then display corresponding menus to admin and volunteer.
- `admin_cli.py` controls user flow for admin. It includes 4 shells: AdminShell, PlanMenu, ManageVolunteerMenu and ManageRefugeeMenu.
- `volunteer_cli.py` controls user flow for volunteer. It only includes VolunteerShell but also have access to ManageRefugeeMenu.
- (merge pending PR#34) `edit_volunteer_cli.py` includes EditVolunteerMenu. I might rename this file and move ManageRefugeeMenu into it as these 2 menus can be accessed by both roles.

### Action Items:
- I haven't finished checking if all the titles and messages are colored/styled. I will update them while testing.
- Check if the volunteer is deactivated when logging in and show corresponding message.

### Testing: 
Run `__main__.py` to test the code.
- Check if the user flow and menu is correct and linked to all functions
- Check if plan/camp/volunteer/refugee can be created/viewed
- Check if the program would not crash due to invalid inputs (error handling)

closes #26 , closes #6
